File: src/main/java/com/will/utils/map/GeoPoint.java
package com.will.utils.map;

/**
 * 经纬度坐标类 支持double和int 输入输出
 * 
 * @author Administrator zwc
 * 
 */
public class GeoPoint
{
	public static String FROM_GEO="84";
	public static String TO_GEO="bd";
	
	public GeoPoint()
	{
	}
	
	public GeoPoint(GeoPoint g)
	{
		if(g==null)
			return;
		
		this.lan=g.getLan();
		this.lon=g.getLon();
		this.direct=g.getDirect();
	}

	public GeoPoint(double lan, double lon)
	{
		this.lan = lan;
		this.lon = lon;
		
	}
	

	public GeoPoint(double lan, double lon, int direct)
	{
		this.lan = lan;
		this.lon = lon;
		this.direct = direct;
		
	}
	
	public GeoPoint(double lan,double lon,boolean transfer)
	{
		this.lan = lan;
		this.lon = lon;
		

	}


	double lan;
	double lon;
	int direct;

	public int getDirect()
	{
		return (direct+360)%360;

	}
	public void setDirect(int direct)
	{
		this.direct = direct;
	}
	public double getLan()
	{
		return lan;
	}

	public void setLan(double lan)
	{
		this.lan = lan;
	}

	public double getLon()
	{
		return lon;
	}

	public void setLon(double lon)
	{
		this.lon = lon;
	}

	public void setLon1E6(int lon)
	{
		this.lon = ((double) lon) / 1e6;
	}

	public int getLan1E6()
	{
		return (int) (lan * 1e6);
	}

	public void setLan1E6(double lan)
	{
		this.lan = ((double) lan) / 1e6;
	}

	public int getLon1E6()
	{
		return (int) (lon * 1e6);
	}
	


}

File: src/main/java/com/will/utils/map/GeoPoint.kt
package com.will.utils.map

/**
 * Created by zoumy on 2017/5/11 15:29.
 */
class GeoPoint() {

    var lan: Double = 0.toDouble()
    var lon: Double = 0.toDouble()
    var direct: Int = 0

    val FROM_GEO = "84"
    val TO_GEO = "bd"

    constructor(point: GeoPoint):this(){
        this.lan = point.lan
        this.lon = point.lon
        this.direct = point.direct
    }

    constructor(lan:Double,lon:Double):this(){
        this.lon = lon
        this.lan = lan
    }

    constructor(lan: Double,lon: Double,direct:Int):this(lan, lon){
        this.direct = direct
    }

}